### PR TITLE
docs: add nextjs overview content; update the format of the quickstarts/nextjs content

### PR DESCRIPTION
**Pull request by alexisintech**
Adds overview content to /references/nextjs/overview.

Adds `<Steps>` to the /quickstarts/nextjs content for better organization and readability.

https://github.com/clerkinc/clerk-docs/assets/98043211/9f81e91e-7c1c-4d24-928e-4df3d4efa8a6

### CODE REVIEW

File: docs/references/nextjs/overview.mdx
Comment (BRKalow) [inline docs/references/nextjs/overview.mdx:1]: Is it right that this has the same content as the quickstart? I don't think we want to duplicate that content here 💭

File: docs/references/nextjs/overview.mdx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+Clerk Next.js also provides client side helpers, such as
+
+- [`getAuth()`](/docs/references/nextjs/get-auth)
+- [`buildClerkProps()`](/docs/references/nextjs/build-clerk-props)
+
+## Other useful resources
+
+- [Authentication Object](/docs/references/nextjs/authentication-object)
+- [Using tRPC](/docs/references/nextjs/trpc)
+- [`withClerkMiddleware()`](/docs/references/with-clerk-middleware) - will soon be deprecated in favor of `authMiddleware()`
```
Comment (BRKalow) [inline docs/references/nextjs/overview.mdx:34]:
```suggestion
## Other useful resources

- [Using Clerk with Next.js & tRPC](/docs/references/nextjs/trpc)
```

File: docs/references/nextjs/overview.mdx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+description: Learn how to use Clerk to quickly and easily add secure authentication and user management to your Next.js application.
+---
+
+# Overview
+
+Clerk makes it simple to add authentication to your Next.js application. Looking to get started? Check out our [getting started guide](/docs/quickstarts/nextjs). If you want to learn more about the capabilities and methods available from Clerk's Next.js SDK, check out the reference content below.
+
+## `authMiddleware()`
+
+The authMiddleware() helper integrates Clerk authentication into your Next.js application through middleware. It's an important step in setting up your application. By default, Clerk's authMiddleware() treats all routes as private if the middleware runs. If you need to make specific routes public, you can learn more [here](/docs/references/nextjs/auth-middleware).
```
Comment (BRKalow) [inline docs/references/nextjs/overview.mdx:12]:
```suggestion
The `authMiddleware()` helper integrates Clerk authentication into your Next.js application through middleware. It allows you to integrate authorization into both the client and server of your application. You can learn more [here](/docs/references/nextjs/auth-middleware).
```

File: docs/references/nextjs/overview.mdx [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+## App Router
+
+Clerk Next.js provides many helpers that are optimized specifically for Next.js's App Router.
```
Comment (BRKalow) [inline docs/references/nextjs/overview.mdx:16]:
```suggestion
Clerk provides first-class support for the [Next.js App Router](https://nextjs.org/docs/app). The below methods and references show how to integrate Clerk features into applications that take advantage of the latest App Router and React Server Components features.
```

File: docs/references/nextjs/overview.mdx [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+- [`auth()`](/docs/references/nextjs/auth)
+- [`currentUser()`](/docs/references/nextjs/current-user)
+- [Route Handlers](/docs/references/nextjs/route-handlers)
+- [Server Actions](/docs/references/nextjs/server-actions)
+
+## Pages Router
+
+Clerk Next.js also provides client side helpers, such as
```
Comment (BRKalow) [inline docs/references/nextjs/overview.mdx:25]:
```suggestion
Clerk continues to provide drop-in support for the Next.js Pages Router. In addition to the main Clerk integration, several methods are available for instrumenting authentication within Pages Router-based applications.
```

**Review by BRKalow — COMMENTED** [review]
Looking good! A few small adjustments

File: docs/quickstarts/javascript.mdx [hunk 5]
```diff
@@ -177,6 +177,8 @@ Now that you have an application integrated with Clerk, you will want to read th
 
     <Cards title="Authentication Components" description="Learn more about all our authentication components." link="/docs/references/javascript/clerk/clerk#components" cta="Learn More" />
 
+    <Cards title="React Hooks" description="Leverage Clerk React hooks for your Next.js application." link="/docs/references/react/use-user" cta="Learn More" />
+
```
Comment (BRKalow) [inline docs/quickstarts/javascript.mdx:181]: Let's remove this one, I don't think it makes sense in the context of the JS quickstart, which doesn't require the use of React or Next.

```suggestion
```

File: docs/quickstarts/nextjs.mdx [hunk 6]
```diff
@@ -22,7 +22,9 @@ By the end of this guide, you will have installed the following user functionali
 - Manage account
 - Sign out
 
-## Install `@clerk/nextjs`
+<Steps>
```
Comment (BRKalow) [inline docs/quickstarts/nextjs.mdx:25]: This looks nice!

**Review by BRKalow — APPROVED** [review]
👏

File: docs/quickstarts/javascript.mdx (hunk 5 above)
Comment (alexisintech) [reply docs/quickstarts/javascript.mdx:181]: I definitely added that to the wrong place!!

**Pull request merged**